R-package/src/install.libs.R

```diff
@@ -142,8 +142,18 @@ if (!use_precompile) {
 dest <- file.path(R_PACKAGE_DIR, paste0("libs", R_ARCH), fsep = "/")
 dir.create(dest, recursive = TRUE, showWarnings = FALSE)
 if (file.exists(src)) {
-  cat("Found library file: ", src, " to move to ", dest, sep = "")
+  print(paste0("Found library file: ", src, " to move to ", dest))
   file.copy(src, dest, overwrite = TRUE)
 } else {
   stop(paste0("Cannot find lib_lightgbm", SHLIB_EXT))
 }
+
+# clean up the "build" directory
+if (dir.exists(build_dir)) {
```

Comment:
`build_dir` variable is not defined when `use_precompile == TRUE`, isn't it?

Reply:
😬 yep you're right!  Let me fix that

Reply:
Moved it up in https://github.com/microsoft/LightGBM/pull/2909/commits/56635a7cf10b53f543db3b96cd1831dd159ff6b8. This also exposed that that `setwd(source_dir)` was unnecessary.

thanks!